scipy/integrate/_cubature.py

```diff
@@ -600,82 +600,74 @@ def __init__(self, f, a, b, xp):
-                self._orig_b[i] = -a[i]
+        # Since we flip the limits, we don't need to separately multiply the
+        # integrand by -1.
+        self._orig_a[inf_end_mask] = -b[inf_end_mask]
+        self._orig_b[inf_end_mask] = -a[inf_end_mask]
 
     @property
     def transformed_limits(self):
         a = xp_copy(self._orig_a)
         b = xp_copy(self._orig_b)
 
-        for index in self._double_inf_pos:
-            a[index] = -1
-            b[index] = 1
+        a[self._double_inf_pos] = -1
+        b[self._double_inf_pos] = 1
 
-        for index in self._semi_inf_pos:
-            a[index] = 0
-            b[index] = 1
+        a[self._semi_inf_pos] = 0
+        b[self._semi_inf_pos] = 1
 
         return a, b
 
     @property
     def points(self):
         # If there are infinite limits, then the origin becomes a problematic point
         # due to a division by zero there.
-        if len(self._double_inf_pos) != 0 or len(self._semi_inf_pos) != 0:
+        if (self._xp.sum(self._double_inf_pos) != 0
+            or self._xp.sum(self._semi_inf_pos) != 0):
             return [self._xp.zeros(self._orig_a.shape)]
         else:
             return []
 
     def inv(self, x):
         t = xp_copy(x)
 
-        for i in self._double_inf_pos:
-            t[..., i] = 1/(x[..., i] + self._xp.sign(x[..., i]))
+        t[:, self._double_inf_pos] = (1/(x[:, self._double_inf_pos]
+                                         + self._xp.sign(x[:, self._double_inf_pos])))
 
-        for i in self._semi_inf_pos:
-            t[..., i] = 1/(x[..., i] - self._orig_a[i] + 1)
+        t[:, self._semi_inf_pos] = 1/(x[:, self._semi_inf_pos]
+                                      - self._orig_a[self._semi_inf_pos] + 1)
 
         return t
 
     def __call__(self, t, *args, **kwargs):
         x = xp_copy(t)
-        jacobian = 1.0
 
-        for i in self._double_inf_pos:
-            # For (-oo, oo) -> (-1, 1), use the transformation x = (1-|t|)/t.
-            x[..., i] = (1 - self._xp.abs(t[..., i])) / t[..., i]
-            jacobian *= 1/(t[..., i] ** 2)
+        # For (-oo, oo) -> (-1, 1), use the transformation x = (1-|t|)/t.
+        x[:, self._double_inf_pos] = ((1 - self._xp.abs(t[:, self._double_inf_pos]))
+                                      / t[:, self._double_inf_pos])
 
-        for i in self._semi_inf_pos:
-            # For (start, oo) -> (0, 1), use the transformation x = start + (1-t)/t.
-            x[..., i] = self._orig_a[i] + (1 - t[..., i]) / t[..., i]
-            jacobian *= 1/(t[..., i] ** 2)
+        # For (start, oo) -> (0, 1), use the transformation x = start + (1-t)/t.
+        x[:, self._semi_inf_pos] = (self._orig_a[self._semi_inf_pos]
+                                    + (1 - t[:, self._semi_inf_pos])
+                                    / t[:, self._semi_inf_pos])
+
+        jacobian_det = self._xp.prod(
+            1/(t[:, self._double_inf_pos | self._semi_inf_pos] ** 2)
```

Comment:
```suggestion
            1/(t[:, self._double_inf_pos | self._semi_inf_pos] ** 2, axis=-1)
```

Adding `axis=-1` gets rid of all non-array API errors. The problem was that `t` has shape `(npoints, ndim)` rather than just `(ndim,)`, so the Jacobian determinant needs to be calculated for each node/evaluation point separately.

Reply:
Great thanks